.github/workflows/ocaml-general.yml

```
@@ -0,0 +1,170 @@
+name: OCaml-General
+on:
+  push:
+    paths:
+      - 'ocaml-general/**'
+      - '!ocaml-general/README.md'
+      - '.github/workflows/ocaml-general.yml'
+    branches: [main]
+
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths:
+      - 'ocaml-general/**'
+      - '!ocaml-general/README.md'
+      - '.github/workflows/ocaml-general.yml'
+
+jobs:
+  validate-ocaml-general:
+    strategy:
+      matrix:
+        ocaml-version:
+          - 4.13.1
+        arch:
+          - amd64
+          # - arm64 # it's taking too long for the purpose of validation
+        node-version:
+          - hydrogen # v18 Active LTS
+        os:
+          - ubuntu.22.04
+    env:
+      DOCKER_TAG: ${{ matrix.os }}-ocaml.${{ matrix.ocaml-version }}-node.${{ matrix.node-version }}-${{ matrix.arch }}
+      CACHE_DIR: /tmp/docker-cache
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up QEMU for arm64
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: arm64
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - run: mkdir -p ${{ env.CACHE_DIR }}
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
```

↓変え忘れました.. 一旦 https://github.com/kxcteam/pubinfra.dockerfiles/actions/runs/4212488051 の結果が出てから修正したい

```suggestion
        uses: actions/cache@v3
```